cli/js/util.ts

```diff
@@ -110,14 +110,9 @@ function pathFromURLPosix(url: URL): string {
 }
 
 export function pathFromURL(pathOrUrl: string | URL): string {
```

Comment:
I think this method attempting to parse strings as URLs makes sense, given the name of the function. (plus, even though this is not part of the public API, returning any string unchanged is sorta weird)

The fix should be in all the methods that use pathFromURL unconditionally, I'll add another comment to where I'm referring

Reply:
@caspervonb I don't know exactly what that reaction means, it could just be disagreement, but in case I didn't explain myself well:

As of this PR, `openSync` (and all fs methods that can take a file URL) takes either a string, which is always a path, or a URL. If it is a URL then `pathFromUrl` will work as expected. If it is a string then it is still passed into `pathFromURL` which does nothing except return it back out.

What my comment was aiming at is that I think that the fs methods should only call `pathFromURL` when passed a URL, since we know passing a string into it won't do anything.

Reply:
> I think this method attempting to parse strings as URLs makes sense

It's path from URL; if its already a path then no just return, no?

> What my comment was aiming at is that I think that the fs methods should only call pathFromURL when passed a URL, since we know passing a string into it won't do anything.

Yeah I just went for removing the offending code rather than starting on a refactoring trip introducing more code.

Reply:
Yeah fair enough on that. I was mostly just concerned that if you just look at the fs methods it looks like we're still parsing everything into paths. I guess it still makes sense this way 👍